File: animals.php
<!-- 
 FILE : animals.php
 PROJECT : PROG2000 - Assignment #2
 PROGRAMMER : lpcSetProg
 FIRST VERSION : 2016-09-10
 DESCRIPTION :
 This application is PHP file, sourced from the BirdToPHP.html file. 
 It was captures the selection option and user name from the html file and 
 inserts it in variables to locate the related files in the directories 
 (NZBirdsTexts, and NZBirdsPhotos). The file extensions are appended to the 
 values in order to display contents. 
 
NOTE: To clarify, birds are creatures under the kingdom called Animalia.
--> 



<html>
<body style="width: 1501px">

<!-- processes the combobox selection [aka bird choice]  -->
<?php $selectOption = $_POST['userSelect']; ?>

<br> 


<!-- append proper file extensions to the values in order to find files  -->

<?php $textFilePath = "./NZBirdsTexts/" . $selectOption . ".txt"; ?> 

<?php $imageFilePath = "./NZBirdsPhotos/" . $selectOption . ".jpg"; ?>


<!-- Testing whether a file exists -->

<?php  
if (file_exists($textFilePath )) 
{ //NOTE: Information is echoed to screen because if it is HTML will execute and display even if file is not found. 
	echo "<p><b> Welcome: </b>". $_POST["name"] . "</p>"; //processes name of user and displays it on screen
	echo "<br><br>";
	echo "<p><b> Your bird: </b>" . $selectOption . "</p><p>&nbsp;</p>"; //displays usage selection
	echo "<p style='width: 426px; height: 194px; float:left'>". file_get_contents($textFilePath) . " </p>"; //displays text file contents
	echo "<img style ='float:left' src='" . $imageFilePath ."' />"; //displays image, to the left of text 

} 
else 
{
echo "The file" . $textFilePath . "does not exist"; //file does not exist 
}
?>


</body>
</html> 

File: cgi_server/animals/animals/animals.cpp
/*
* FILE : animals.cpp
* PROJECT : PROG2000 - Web Design - Assignment #2
* PROGRAMMERS :  lpcSetProg
* FIRST VERSION : 2016-10-09
* DESCRIPTION :
* This file contains the CGI script to handle the query string passed from 
* Bird_get.html. Query string from html file is displayed as ex. "name=Lev&userSelect=Kiwi"
* The CGI program acts as a server to display HTML script in a browser. There is a related PHP server (animals.php)
* which processes Bird_post.html. On the server, two directories exist which hold the text descriptions and image files. 
*/


#include "animals.h"

int main()
{
	//Variables
	string birdImageContents;
	string birdImageFile;
	string birdTextFile;
	string birdTextContents;
	string fileExtentionImage;
	string fileExtentionText; 
	string userName;
	string userSelect;
	string QueryString;
	int position1 = 0; //used for find positions of delimiters 
	int position2 = 0;
	int position3 = 0;


	//Query string from html file is displayed as ex. "name=Lev&userSelect=Kiwi" 
	//The two form values are separated by an ampersand (&).
	//Researched information at : http://www.cgi101.com/book/ch3/text.html
	//Parse string by location of delemiters '=' and '&' and find the 'userName' and 'userSelect' from combobox 
	QueryString = getenv("QUERY_STRING");
	position1 = QueryString.find_first_of('=');
	position2 = QueryString.find_first_of('&');
	position3 = QueryString.find_last_of('=');
	userName = QueryString.substr(++position1, position2 - position1);  //assign
	userSelect = QueryString.substr(++position3);


	//Locate matching text based on user selection 
	fileExtentionText = ".txt";
	birdTextFile  = userSelect + fileExtentionText;	//append .txt file extention in order to find related text file
	birdTextContents = TextLocator(birdTextFile);


	//Can be same file 
	//Locate matching image based on user selection 
	fileExtentionImage = ".jpg"; //append .jpg file extention in order to find related image 
	birdImageFile = userSelect + fileExtentionImage;
	birdImageContents = ImageLocator(birdImageFile);
	


	//MAIN CGI DISPLAY 
	cout << "Content-type:text/html\r\n\r\n"; //signifies where CGI script begins 
	cout << "<html>\n";
	cout << "<head>\n";
	cout << "<title>New Zealand Native Bird Conservatory</title>\n"; ///hello user name!
	cout << "</head>\n";
	cout << "<body>\n";
	cout << "<p> <b> Welcome: " << userName << " </b> </p>\n "; //<< endl; //break line 
	cout << "<p> <b> Your bird is : " << userSelect << " </b> </p>\n "; //<< endl; //break line 
	cout << "<p><span style='float:left; width: 426px;'>  " << birdTextContents << "</span><img src = '" << birdImageContents << "'" << " / >" << "</p>\n"; //aligns image with text
	cout << "</body>\n";
	cout << "</html>\n"; 



	///These strings were used for testing to make sure data from queryString was parsed correctly 
	//cout << userName;
	//cout << userSelect;
	//cout << QueryString;
	//cout << "<img src='" << birdImageContents << "'" << "/>" << endl;

	return 0;
}







